app/connectors/NsiConnector.scala

```diff
@@ -93,7 +93,7 @@ class NsiConnector @Inject() (
     val domain       = servicesConfig.baseUrl(serviceName)
     val rootPath     = servicesConfig.getString(s"microservice.services.$serviceName.rootPath")
     val resourcePath = servicesConfig.getString(s"microservice.services.$serviceName.$endpoint")
-    val pathParams   = params.map("/" + URLEncoder.encode(_, "UTF-8")).mkString
+    val pathParams   = params.map("/" + URLEncoder.encode(_, "UTF-8")).mkString.replaceAll("\\+","%20")
```

Comment:
For clarity for whoever is going to work on this piece of code in the future, could you extract the encoding part into a private method with appropriate name? I'm concerned that leaving this in its current form is not showing the intent clearly. And in that case I'd call `replaceAll` for each encoded parameter. Not for all at the end.